lib/index.ts

```diff
@@ -1,13 +1,34 @@
 import maps, { ColorMap, MapKey } from './maps'
 import { hexColorToRGBA, lerpRGBA, RGBA } from './util'
 
-export type Palette = string[] | RGBA[] | number[]
 export type ColorMapInput = MapKey | ColorMap
 export type PaletteFormat = 'float' | 'rgba' | 'cssHex' | 'cssRGBA' | 'number'
 type CreatePaletteOptions = {
   map?: ColorMapInput,
   steps?: number,
-  format?: PaletteFormat
+}
+
+class Palette {
+  private readonly colors: RGBA[]
+
+  constructor(colors: RGBA[]) {
+    this.colors = colors
+  }
+
+  /** Convert this palette into a specified format */
+  format(_format: 'cssHex'): string[]
+  format(_format: 'cssRGBA'): string[]
+  format(_format: 'number'): number[]
+  format(_format: 'float'): number[]
+  format(_format: 'rgba'): RGBA[]
+  format(format: PaletteFormat) {
```

Comment:
The format option wasn't required in the previous API, can we make it optional here too so calling format without providing a format will use the default?